File: Explorer/MicrosoftOrganizationChartReader/Records/LevelRecord.cs
using System.ComponentModel;
using System.IO;

namespace OrganizationChartReader
{
    public class LevelRecord
    {
        public ushort Size { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown1 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown2 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown3 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown4 { get; set; }

        public ushort Count { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown5 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public ushort Unknown6 { get; set; }

        [TypeConverter(typeof(HexTypeConverter))]
        public byte Unknown7 { get; set; }

        public LevelRecord(BinaryReader reader)
        {
            Size = reader.ReadUInt16();
            Unknown1 = reader.ReadUInt16();
            Unknown2 = reader.ReadUInt16();
            Unknown3 = reader.ReadUInt16();
            Unknown4 = reader.ReadUInt16();
            Count = reader.ReadUInt16();
            Unknown5 = reader.ReadUInt16();
            Unknown6 = reader.ReadUInt16();
            Unknown7 = reader.ReadByte();
        }
    }
}

File: Sources/MicrosoftOrganizationChartReader/Records/LevelRecord.swift
//
//  LevelRecord.swift
//
//
//  Created by Hugh Bellamy on 25/01/2021.
//

import DataStream

public struct LevelRecord {
    public let size: UInt16
    public let unknown1: UInt16
    public let unknown2: UInt16
    public let unknown3: UInt16
    public let unknown4: UInt16
    public let count: UInt16
    public let unknown5: UInt16
    public let unknown6: UInt16
    public let unknown8: UInt8
    
    public init(dataStream: inout DataStream) throws {
        self.size = try dataStream.read(endianess: .littleEndian)
        self.unknown1 = try dataStream.read(endianess: .littleEndian)
        self.unknown2 = try dataStream.read(endianess: .littleEndian)
        self.unknown3 = try dataStream.read(endianess: .littleEndian)
        self.unknown4 = try dataStream.read(endianess: .littleEndian)
        self.count = try dataStream.read(endianess: .littleEndian)
        self.unknown5 = try dataStream.read(endianess: .littleEndian)
        self.unknown6 = try dataStream.read(endianess: .littleEndian)
        self.unknown8 = try dataStream.read()
    }
}
